Auto-generate docs in dev.loklak.org repository

imujjwal96 opened the pull request:
Related issue: #1109 
### Short description
I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #1109 commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

yukiisbored commented:
Actually, I'm planning to push the raw rST/md files to the repository and let it build it via CI rather than pushing the compiled HTML pages so everything would be integrated nicely in Sphinx.

imujjwal96 commented:
@yukiisbored  If you want, then I could assist you with it.

mariobehling commented:
@imujjwal96 Yes, please follow this approach of @yukiisbored. Then what should happen to this PR?

imujjwal96 commented:
What I did:
1. Removed docs directory.
2. Changed the update-docs-repo.sh such that the raw README.rst file is transferred to dev.loklak.org/raw/ directory aas server.rst

Similarly,  all the raw file from their repositories can be copied from their repositories to the dev.loklak.org/raw directory. In the CI of dev.loklak.org repository, we can generate the documentation via sphinx

pri22296 commented:
@imujjwal96 I don't think the docs directory should be removed. Only sphinx related files should be removed(conf.py, Makefile, etc). The actual documentation files should be kept in this repo. They would be pushed to dev.loklak.org via a script.

Also in update-docs-repo, you should clone the subtree corresponding to this repo, not the gh-pages branch.

mariobehling commented:
@imujjwal96 Yes, exactly documentation should still stay in the docs folder of repositories.

sch00lb0y commented:
@imujjwal96 kindly check this after sphnix generation copy the /_build/html/ to the doc kindly check this script for example https://github.com/sch00lb0y/yaydoc/blob/master/generate.sh

imujjwal96 commented:
This PR is messed up. Shall I give a new PR?

imujjwal96 commented:
Hey @pri22296 I am not well versed with subtrees. Can you please give an excerpt of it and its significance in the current issue.

pri22296 commented:
@imujjwal96 In the dev.loklak.org repository,  documentation for different loklak repos are stored in different subtrees. The script needs to clone the subtree specific to the current repo. Take a look at this [script](https://github.com/loklak/dev.loklak.org/blob/gh-pages/.ci/pull.sh) by @yukiisbored from the dev.loklak.org repository.

imujjwal96 commented:
@pri22296 There we are pulling the complete content of the branch. What I intend to do now is to add only a specific folder, ie., docs/ folder to the dev.loklak.org repository

pri22296 commented on .utility/update-docs-repo.sh:
```diff
@@ -1,28 +1,29 @@
-echo "Preparing git ..."
-git config --global user.email "travis@travis-ci.org"
-git config --global user.name "travis-ci"
+git clone --quiet --branch=gh-pages git@github.com:loklak/dev.loklak.org.git gh-pages
```
Don't clone gh-pages branch. It will contain the built documentation. Clone master instead

imujjwal96 replied:
Or should we create a separate branch. Master already has some preexisting code.
I am not sure if that aligns with the changes i am making.

pri22296 commented on .utility/update-docs-repo.sh:
```diff
@@ -1,28 +1,29 @@
+cp -f $HOME/README.rst .
+git add -f .
+git commit -m "Latest server documentation file on successful travis build $TRAVIS_BUILD_NUMBER auto-pushed to dev.loklak.org"
+git push -fq origin gh-pages > /dev/null 2>&1
```
Don't push to gh-pages. push to master. These are raw md/rst files not html

imujjwal96 commented:
@mariobehling @yukiisbored @pri22296 @sch00lb0y 
If you all could review the changes further, I would make the required changes.
Don't want to push multiple commits with small changes.

sch00lb0y commented:
@imujjwal96 if you change what @pri22296 mentioned it'll work fine and test your script on local rep before sending PR

imujjwal96 commented:
in https://github.com/loklak/dev.loklak.org/blob/1796511d5f1ee18df974370f48bf332584e9ca2d/.ci/pull.sh 
the code pulls the subtree directory without adding one. When is the subtree added? Am I missing something?

mariobehling commented:
@imujjwal96 Please get together with your co-developers and solve the open questions. PRs/issues should not usually take longer than a day. The flow is very slow here. We need to find better ways. Please take the first step and get your co-developers to review this.

imujjwal96 commented:
@pri22296 @sch00lb0y
Can you please review this?

yukiisbored commented on .directory:
```diff
@@ -0,0 +1,6 @@
+[Dolphin]
```
Please remove this file

sch00lb0y commented on .utility/update-docs-repo.sh:
```diff
@@ -1,28 +1,29 @@
-./.ci/pull.sh config.csv
+cp -f $HOME/README.rst .
+git add -f .
+git commit -m "Latest server documentation file on successful travis build $TRAVIS_BUILD_NUMBER auto-pushed to dev.loklak.org"
```
commit message some thing like this
```bash
git commit -m "[Auto] Update Built Docs ($(date +%Y-%m-%d.%H:%M:%S))"
```

imujjwal96 replied:
I've kept it the way it was originally. Do I really need to change it?

sch00lb0y replied:
okay keep like that only

imujjwal96 commented:
Should I cherry-pick specific commits to a new branch and give a new PR?

yukiisbored commented:
@imujjwal96 Github only cares what on the branch name, just create a new branch from your current one. Destroy the PR branch, Create a new one with the same name, Cherry pick commits from the other branch, and force push.

imujjwal96 commented:
@yukiisbored it broke.
